src/GrafanaDashboardDeploy/src/DeployTool.cs

```diff
@@ -0,0 +1,152 @@
+                    }
+
+                    secretValue = await GetSecretAsync(secretName).ConfigureAwait(false);
+                    secretData.Value = secretValue;
+                }
+
+                await grafanaClient.CreateDatasourceAsync(datasource.ToObject<JObject>()).ConfigureAwait(false);
+            }
+
+            foreach (var dashboard in data["dashboards"])
+            {
+                string dashboardFolderUid = dashboard["meta"]["folderUid"].Value<string>();
+
+                int folderId = folderIdMap[dashboardFolderUid];
+
+                await grafanaClient.CreateDashboardAsync(dashboard["dashboard"].ToObject<JObject>(), folderId).ConfigureAwait(false);
+            }
+        }
+
+        public static async Task MakeExportPack(GrafanaClient grafanaClient, IEnumerable<string> dashboardUids, string outputFilePath)
+        {
+            var dashboards = new List<JObject>();
+            var folders = new List<JObject>();
+            var dataSources = new List<JObject>();
+
+            foreach (var dashboardUid in dashboardUids)
+            {
+                // Get a dashboard
+                JObject dashboard = await grafanaClient.GetDashboardAsync(dashboardUid).ConfigureAwait(false);
+
+                // Cache dashboard data needed for post
+                JObject slimmedDashboard = Util.SanitizeDashboard(dashboard);
+
+                // Get the dashboard folder data and cache if not already present
+                int folderId = Util.ExtractFolderId(dashboard);
+                var folder = await grafanaClient.GetFolder(folderId).ConfigureAwait(false);                
+
+                // Cache if not already present
+                if (!folders.Any(v => v.SelectToken("$.uid").Value<string>() == folder["uid"].Value<string>()))
+                {
+                    var slimmedFolder = Util.SanitizeFolder(folder);
+                    folders.Add(slimmedFolder);
+                }
+
+                // Folder uid is needed for the dashboard export object                    
+
+                var dashboardObject = Util.ConstructDashboardExportObject(slimmedDashboard, folder["uid"].Value<string>());
+                dashboards.Add(dashboardObject);
+
+                // Get datasources used in the dashboard
+                var dataSourceNames = Util.ExtractDataSourceNames(dashboard);
+
+                foreach (var datasourceName in dataSourceNames)
+                {
+                    // Cache if not already present
+                    if (!dataSources.Any(v => v.SelectToken("$.name").Value<string>() == datasourceName))
+                    {
+                        var dataSource = await grafanaClient.GetDataSource(datasourceName).ConfigureAwait(false);
+                        dataSource = Util.SanitizeDataSource(dataSource);
+                        dataSources.Add(dataSource);
+                    }
+                }
+            }
+
+            JObject output = new JObject();
```

Comment:
Since there are only 3 static values, you can do the intilializer thing
```csharp
var output = new JObject{
  ["folders"] = new JArray(folders),
  //...
```